packages/fx-core/src/component/generator/officeAddin/generator.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  FxError,
+  Inputs,
+  Result,
+  ok,
+  err,
+  ManifestUtil,
+  devPreview,
+  ContextV3,
+} from "@microsoft/teamsfx-api";
+import { join } from "path";
+import {
+  AddinNameQuestion,
+  AddinLanguageQuestion,
+  OfficeHostQuestion,
+  getTemplate,
+  AddinProjectFolderQuestion,
+  AddinProjectManifestQuestion,
+} from "./question";
+import { helperMethods } from "./helperMethods";
+import { OfficeAddinManifest } from "office-addin-manifest";
+import projectsJsonData from "./config/projectsJsonData";
+import * as childProcess from "child_process";
+import { promisify } from "util";
+import { CopyFileError } from "../../../core/error";
+import _ from "lodash";
+import { hooks } from "@feathersjs/hooks/lib";
+import { ActionExecutionMW } from "../../middleware/actionExecutionMW";
+import { Generator } from "../generator";
+
+const childProcessExec = promisify(childProcess.exec);
+
+const componentName = "office-addin";
+const telemetryEvent = "generate";
+const templateName = "office-addin";
+
+export class OfficeAddinGenerator {
+  @hooks([
+    ActionExecutionMW({
+      enableTelemetry: true,
+      telemetryComponentName: componentName,
+      telemetryEventName: telemetryEvent,
+      errorSource: componentName,
+    }),
+  ])
+  static async generate(
+    context: ContextV3,
+    inputs: Inputs,
+    destinationPath: string
+  ): Promise<Result<undefined, FxError>> {
+    const result = await OfficeAddinGenerator.doScaffolding(context, inputs, destinationPath);
+    if (result.isErr()) {
+      return err(result.error);
+    }
+
+    const templateRes = await Generator.generateTemplate(
+      context,
+      destinationPath,
+      templateName,
+      "ts"
```

Comment:
There is a language question that should be used to set this value.  Currently it is only Typescript (based on the settings json) and so the question is skipped in the UI but setting it up now allow for future changes to be easier.